packages/genesys-spark-components/src/components/beta/gux-avatar/gux-avatar.tsx

```diff
@@ -0,0 +1,173 @@
+import { Component, h, JSX, Prop, Element, Fragment } from '@stencil/core';
+import { trackComponent } from '@utils/tracking/usage';
+import {
+  GuxAvatarStatus,
+  GuxAvatarSize,
+  GuxAvatarAccent
+} from './gux-avatar.types';
+import { hasSlot } from '@utils/dom/has-slot';
+import { logWarn } from '@utils/error/log-error';
+
+/**
+ * @slot image - Headshot photo.
+ */
+
+@Component({
+  styleUrl: 'gux-avatar.scss',
+  tag: 'gux-avatar-beta',
+  shadow: true
+})
+export class GuxAvatar {
+  @Element()
+  root: HTMLElement;
+
+  @Prop()
+  status: GuxAvatarStatus = 'available';
```

Comment:
Thoughts on not hardcoding string values like above but using some sort of typed list of pre-defined values?